gems/aws-sdk-core/lib/aws-sdk-core/assume_role_credentials.rb

```diff
@@ -66,7 +66,7 @@ def refresh
       creds = c.credentials
       account_id =
         begin
-          ARNParser.parse(c.assumed_role_user.arn).account_id
+          ARNParser.parse(c.assumed_role_user&.arn).account_id
```

Comment:
Don't you need this on assume role web identity credentials? It would be nice if you can make a private method for parsing both of those things to make the code nicer.

Reply:
yeah, good catch. Cleaned up both.